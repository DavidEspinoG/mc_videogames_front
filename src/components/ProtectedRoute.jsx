import { PropTypes } from 'prop-types';
import { Navigate, Outlet } from 'react-router-dom';

const ProtectedRoute = ({ isAllowed, redirectPath, children }) => {
  if (!isAllowed) {
    return <Navigate to={redirectPath} replace />;
  }

  return children ?? <Outlet />;
};

ProtectedRoute.propTypes = {
  isAllowed: PropTypes.bool.isRequired,
  redirectPath: PropTypes.string,
  children: PropTypes.node,
};

ProtectedRoute.defaultProps = {
  redirectPath: '/',
  children: null,
};

export default ProtectedRoute;
