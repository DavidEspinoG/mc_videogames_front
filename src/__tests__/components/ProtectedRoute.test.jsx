import ProtectedRoute from '../../components/ProtectedRoute';
import renderWithProviders from '../test-utils';

describe('ProtectedRoute', () => {
  const isAllowed = true;

  it('should render as expected', () => {
    const { container } = renderWithProviders(
      <ProtectedRoute isAllowed={isAllowed}>Hello</ProtectedRoute>,
    );
    expect(container).toMatchSnapshot();
  });
});
