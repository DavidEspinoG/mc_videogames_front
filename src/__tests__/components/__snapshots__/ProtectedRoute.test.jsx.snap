// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProtectedRoute > should render as expected 1`] = `
<div>
  Hello
</div>
`;
